cmd/soroban-rpc/internal/network/requestDurationLimiter.go

```diff
@@ -132,14 +135,36 @@ func (q *httpRequestDurationLimiter) ServeHTTP(res http.ResponseWriter, req *htt
 	if q.limitThreshold != time.Duration(0) {
 		limitCh = time.NewTimer(q.limitThreshold).C
 	}
-	requestCompleted := make(chan interface{}, 1)
+	requestCompleted := make(chan []string, 1)
 	requestCtx, requestCtxCancel := context.WithTimeout(req.Context(), q.limitThreshold)
 	defer requestCtxCancel()
 	timeLimitedRequest := req.WithContext(requestCtx)
 	responseBuffer := makeBufferedResponseWriter(res)
 	go func() {
+		defer func() {
```

Comment:
this seems like a general panic recover utility method, is it worth pulling this inline func up to a package private method instead like `handlePanic(finally chan []string)`, might help readability here?

Reply:
I think that it makes sense. Let me merge https://github.com/stellar/soroban-tools/pull/856 first, and I'll rebase this & refactor to implement that change.